yarn-project/simulator/src/public/public_db_sources.ts

```diff
@@ -322,11 +279,49 @@ export class WorldStateDB implements CommitmentsDB {
   public async getNullifierIndex(nullifier: Fr): Promise<bigint | undefined> {
     const timer = new Timer();
     const index = await this.db.findLeafIndex(MerkleTreeId.NULLIFIER_TREE, nullifier.toBuffer());
-    this.log.debug(`[DB] Fetched nullifier index`, {
+    this.logger.debug(`[DB] Fetched nullifier index`, {
       eventName: 'public-db-access',
       duration: timer.ms(),
       operation: 'get-nullifier-index',
     } satisfies PublicDBAccessStats);
     return index;
   }
+
+  /**
+   * Commit the pending changes to the DB.
+   * @returns Nothing.
+   */
+  commit(): Promise<void> {
+    for (const [k, v] of this.checkpointedWriteCache) {
+      this.committedWriteCache.set(k, v);
+    }
+    // uncommitted writes take precedence over checkpointed writes
+    // since they are the most recent
+    for (const [k, v] of this.uncommittedWriteCache) {
+      this.committedWriteCache.set(k, v);
+    }
+    return this.rollbackToCommit();
+  }
```

Comment:
Only weirdness here is that commit/rollback/etc is only relevant to a subset of the DB (public storage only). But I think this is fine. Maybe they could be renamed commitPublicStorage or something like that, but I won't block this PR on that.

Reply:
good notes